src/System.Device.Gpio/System/Device/Spi/Devices/Windows10SpiDevice.cs

```diff
@@ -112,7 +123,17 @@ public override void WriteByte(byte value)
         /// </param>
         public override void Write(ReadOnlySpan<byte> buffer)
         {
-            _winDevice.Write(buffer.ToArray());
+            if (_isInverted)
+            {
+                Span<byte> toSend = stackalloc byte[buffer.Length];
+                buffer.CopyTo(toSend);
+                ReverseByte(toSend);
+                _winDevice.Write(toSend.ToArray());
```

Comment:
You can simplify this whole method to:

```csharp
byte[] toSend = buffer.ToArray();
if (_isInverted)
{
    ReverseByte(toSend);
}

_winDevice.Write(toSend);
```

Reply:
ho yes and the rest as well. Perfect catch!